plugins/nf-nomad/src/main/nextflow/nomad/executor/NomadService.groovy

```diff
@@ -64,24 +67,25 @@ class NomadService implements Closeable{
     String submitTask(String id, String name, String image,
                       List<String> args,
                       String workingDir,
-                      Map<String, String>env){
+                      Map<String, String>env,
+                      Resources resources){
         Job job = new Job();
         job.ID = id
         job.name = name
         job.type = "batch"
         job.datacenters = this.config.jobOpts.datacenters
         job.namespace = this.config.jobOpts.namespace
 
-        job.taskGroups = [createTaskGroup(id, name, image, args, workingDir, env)]
+        job.taskGroups = [createTaskGroup(id, name, image, args, workingDir, env, resources)]
 
         JobRegisterRequest jobRegisterRequest = new JobRegisterRequest();
         jobRegisterRequest.setJob(job);
         JobRegisterResponse jobRegisterResponse = jobsApi.registerJob(jobRegisterRequest, config.jobOpts.region, config.jobOpts.namespace, null, null)
         jobRegisterResponse.evalID
     }
 
-    TaskGroup createTaskGroup(String id, String name, String image, List<String> args, String workingDir, Map<String, String>env){
-        def task = createTask(id, image, args, workingDir, env)
+    TaskGroup createTaskGroup(String id, String name, String image, List<String> args, String workingDir, Map<String, String>env, Resources resources){
```

Comment:
don't like to expose specific nomad objects outside of the service

I mean, Resources is something strong related with the Service but not with the Task. Maybe we can use some TaskResource class as parameter and let the Service do the conversion to a nomad Resource object

it sounds a little more complicated/elaborated but in this way we have separate both concepts (nextflow vs nomad)

Reply:
Hi @jagedn , okay so I've tried to refactor the code keeping different layers separate. Let me know what you think about the current structure?


I do think that we can explore the idea regarding a specific "translation" class which  covers all aspects of `NomadJobDefinition` from a `NextflowTask`, however I suggest that we still continue the current iterations and revisit this once the plugin has stabilised in its design - what do you think?